src/main/java/org/sunbird/operationalreports/service/OperationalReportServiceImpl.java

```diff
@@ -359,14 +376,45 @@ private Map<String, Object> getSearchObject(List<String> userIds) {
         return requestObject;
     }
 
-    private long getParsedDate(String reportExpiryDate) throws ParseException {
-        long reportExpiryDateMillis;
-        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
-        Date parsedDate = dateFormat.parse(reportExpiryDate);
-        reportExpiryDateMillis = parsedDate.getTime();
-        return reportExpiryDateMillis;
+    public SBApiResponse readGrantAccess(String authToken,String userOrgId) {
```

Comment:
userOrgId -- should not be coming as input from controller... 

read userId from access token, call userutilityserviceImpl.getUserDetailsFromDB -- from the user record get the rootOrgId value...